### PR TITLE
Use `_repr_json_` rather than `_repr_html_`

### PR DESCRIPTION
**Pull request by jsignell**
**Related Issue(s):**

- #1097, #1124

**Description:**

This PR is pretty extreme, but basically it rips out all the jinja templates and `_repr_html_` methods and instead uses the ipython json display mechanism. To me this feels more like what I want to see as the display for a STAC object :shrug: 

Screenshot from 2023-05-23 10-36-49

**Pros:**
- Cuts a lot of code
- Size of saved notebooks is smaller
- No extra GETs (#1097)
- Potentially closer to what the user is used to for STAC objects?

**Cons:**
- Falls back to `_repr_` in static view: https://github.com/stac-utils/pystac/blob/21b747e62e72b4906703ba91397bfc1ad6b1630c/docs/quickstart.ipynb
- Falls back to `_repr_` in classic notebook
- Change from existing look and feel.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `scripts/test`)
- [ ] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

**Comment by pjhartzell** [comment]
I _really_ like this. Complete information, familiar format, less code.

**Review by gadomski — COMMENTED** [review]
I'm a little notebook-slow, how do I get the nicely-formated JSON when I run (e.g.) the quickstart notebook locally? I just tried things out and it looks like it's just printing the dict:

Screenshot 2023-05-25 at 9 26 02 AM

File: pystac/item_collection.py
[the remark this answers is not in the document]
Comment (jsignell) [reply pystac/item_collection.py:1]: I thought it was getting it via inheritance

**Comment by jsignell** [comment]
> I'm a little notebook-slow, how do I get the nicely-formated JSON when I run (e.g.) the quickstart notebook locally? I just tried things out and it looks like it's just printing the dict:
> Screenshot 2023-05-25 at 9 26 02 AM

Assuming you are in jupyter lab, you don't want the `to_dict()`. Just type `asset` in a cell and execute it

**Comment by gadomski** [comment]
Hm, I'm just in the VSCode jupyter interface, and it looks like that doesn't do the pretty printing either. Is there some sort of extension I need to enable?

Screenshot 2023-05-25 at 10 55 17 AM

I think I can state a design goal: a vanilla install of our doc dependencies has pretty-printing in a local notebook, e.g. in VSCode.

**Comment by jsignell** [comment]
Oh that is such a bummer. It looks like vscode hasn't implemented `IPython.display.JSON` :cry: 

image

**Comment by gadomski** [comment]
Bummer indeed. I like the format you showed originally much better than the html. I'm a bit reluctant to just rip the HTML out, since so many folks use VSCode (and there may be other common environments where HTML is supported but JSON is not). I guess we could keep the html and just _add_ json?

**Comment by jsignell** [comment]
Yeah... I guess we could roll our own html that looks like the output of `IPython.display.JSON` and fall back to that on vscode. It would just take a bit longer to implement so I'd probably need someone to actually say they want that :)

**Comment by jsignell** [comment]
Closing this for now since there is no built-in vscode support and it would take a non-trivial amount of work to implement a pretty json jinja template.

**Pull request closed without merging**